evap/staff/importers.py

```diff
@@ -417,6 +464,83 @@ def process(cls, excel_content, test_run):
+        if file_content:
+            user_list, self.success_messages, self.warnings, self.errors = UserImporter.process(file_content, test_run)
+        elif source_course:
+            user_list = list(source_course.participants.all())
+
+        usernames = [user.username for user in user_list]
+        already_related = course.participants.filter(username__in=usernames).all()
+
+        if already_related:
+            msg = _("The following {} user(s) are already course participants in course {}:").format(len(already_related), course.name)
+            msg += create_user_list_string_for_message(already_related)
+
+        # since the user profiles are not necessarily saved to the database, they are not guaranteed to have a pk yet which
+        # makes anything relying on hashes unusable here (for a faster list difference)
+        users_to_add = [user for user in user_list if user not in already_related]
+
+        if test_run:
+            msg = _("{} participants would be added to the course {}:").format(len(users_to_add), course.name)
+            msg += create_user_list_string_for_message(users_to_add)
+        else:
+            course.participants.add(*users_to_add)
+            msg = _("{} Participants added to course {}").format(len(users_to_add), course.name)
+            msg += create_user_list_string_for_message(users_to_add)
+
+        self.success_messages.append(mark_safe(msg))
+
+    def process_contributors(self, course, test_run, file_content=None, source_course=None):
+        if file_content:
+            user_list, self.success_messages, self.warnings, self.errors = UserImporter.process(file_content, test_run)
+        elif source_course:
+            user_list = list(UserProfile.objects.filter(contributions__course=source_course))
+
+        already_related = Contribution.objects.filter(course=course, contributor__in=user_list).all()
+        if already_related:
+            msg = _("The following {} user(s) are already contributing to course {}:").format(len(already_related), course.name)
+            msg += create_user_list_string_for_message(already_related)
+
+        # since the user profiles are not necessarily saved to the database, they are not guaranteed to have a pk yet which
+        # makes anything relying on hashes unusable here (for a faster list difference)
+        users_to_add = [user for user in user_list if user not in already_related]
+
+        if test_run:
+            msg = _("{} contributors would be added to the course {}:").format(len(users_to_add), course.name)
+            msg += create_user_list_string_for_message(users_to_add)
+        else:
+            for user in users_to_add:
+                order = Contribution.objects.filter(course=course).count()
+                contribution = Contribution(course=course, contributor=user, order=order)
+                contribution.save()
+                course.contributions.add(contribution)
```

Comment:
check out https://docs.djangoproject.com/en/1.11/ref/models/querysets/#create. also, the last line shouldn't be necessary, because you already told the contribution it belongs to that course.